packages/gatsby-theme-newrelic/src/utils/createTessen.js

```diff
@@ -33,6 +35,24 @@ const tessenAction =
       return warnAboutNoop({ config, action, name, category });
     }
 
+    if (!CAMEL_CASE.test(name)) {
+      return warning(
+        false,
+        `tessen.${action}: The 'name' argument needs to be in camelCase. This has resulted in a noop. Please change '${name}' to something like '${convertToCamelCase(
+          name
+        )}'.`
+      );
+    }
+
+    if (!TITLE_CASE.test(category)) {
+      return warning(
+        TITLE_CASE.test(category),
+        `tessen.${action}: The 'category' argument needs to be in TitleCase. This has resulted in a noop. Please change '${category}' to something like '${convertToTitleCase(
+          category
```

Comment:
i like the intent here, but this may be a bit much for just letting us know that our fields are incorrectly formatted. 

a particular edge case we introduce with this is if we accidentally supply null, undefined, or data of an incorrect type, this will throw an error.

we could rip this out, or update the functions to be more robust, but i think as long as the warning tells us the value needs to be TitleCase or CamelCase thats enough.